drivers/project_scorecard/app/controllers/project_scorecard/warehouse_reports/scorecards_controller.rb

```diff
@@ -0,0 +1,138 @@
+###
+# Copyright 2016 - 2020 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module ProjectScorecard::WarehouseReports
+  class ScorecardsController < ApplicationController
+    include WarehouseReportAuthorization
+    include ArelHelper
+    # TODO: wjat are the access rules?
+    before_action :set_projects, :set_current_reports
+    before_action :set_report, only: [:show, :edit, :update]
+
+    def index
+      start_date = Date.current.prev_month.beginning_of_month
+      end_date = Date.current.prev_month.end_of_month
+
+      @range = ::Filters::FilterBase.new(start: start_date, end: end_date)
+    end
+
+    def for_project
+      project = project_scope.find(params[:project_id].to_i)
+      @name = project.name
+      @reports = reports_scope.where(project_id: project.id)
+      render :report_list
+    end
+
+    def show
+      report_id = { report_id: @report.id }
+      @pdf_export = ProjectScorecard::DocumentExports::ScorecardExport.new(query_string: report_id.to_query)
+    end
+
+    def create
+      @generate = create_params[:generate] == '1'
+      @email = create_params[:email] == '1'
+
+      @range = ::Filters::FilterBase.new
+      @range.set_from_params(filter_params)
+      errors = @range.errors.messages.map { |k, v| "#{k}: #{v.join(', ')}".humanize }
+
+      @project_ids = params[:project]&.keys&.map(&:to_i) || []
+      @project_ids = project_scope.where(id: @project_ids).pluck(:id) # viewability allowlist
+
+      errors << 'At least one project must be selected' if @project_ids.empty?
+
+      if errors.any?
+        flash[:error] = errors.join('<br />'.html_safe)
+      elsif @generate
+        generate_for_projects(@project_ids, @email, @range, current_user)
+      elsif @email
+        email_to_projects(@project_ids)
+      end
+
+      render action: :index
+    end
+
+    def edit
+    end
+
+    def update
+      if params[:commit] == 'Save & Complete'
+        case @report.status
+        when 'pre-filled'
+          @report.update!(status: 'ready')
+        when 'ready'
+          @report.update!(status: 'completed')
+        end
+      end
+      @report.update!(scorecard_params)
+      redirect_to action: :show
+    end
+
+    private def generate_for_projects(ids, send_email, range, user)
+      ids.each do |id|
+        report = reports_scope.create(project_id: id, user_id: user.id, start_date: range.first, end_date: range.last)
+        ProjectScorecard::PopulateScorecard.perform_later(report.id, send_email, user.id)
+      end
+    end
+
+    private def email_to_projects(ids)
```

Comment:
I wonder if we should try an use the project/organization contacts for the recipients.